pkg/store/sqlite/search.go

```diff
@@ -62,11 +66,43 @@ func (s *Store) SearchCounterparties(ctx context.Context, query *models.SearchQu
 		}
 
 		stub.distance = fuzzy.RankMatchNormalizedFold(query.Query, stub.name)
-		if stub.distance < 0 {
-			continue
+		if stub.distance >= 0 {
+			stubs = append(stubs, stub)
 		}
+	}
 
-		stubs = append(stubs, stub)
+	// If the query is a URL candidate, then check if it matches a website or endpoint
+	// of any of the counterparties in the database and add them to the result stubs.
+	// NOTE: deduplication is necessary in the case that the query matches both a
+	// website and the name of the counterparty.
+	if domainParam := NormURL(query.Query); domainParam != "" {
+		log.Debug().Str("domain", domainParam).Msg("counterparty search website/endpoint match")
+		var domainRows *sql.Rows
+		if domainRows, err = tx.Query(counterpartyDomainSearchSQL, sql.Named("domainParam", "%"+domainParam+"%")); err != nil {
+			return nil, dbe(err)
+		}
+		defer domainRows.Close()
+
+	domainScan:
+		for domainRows.Next() {
+			stub := counterpartyStub{}
+			if err = domainRows.Scan(&stub.id, &stub.name); err != nil {
+				return nil, err
+			}
+
+			// Check if the stub is already in the list of stubs
+			for _, s := range stubs {
+				if s.id == stub.id {
+					continue domainScan // already in the list, skip it
+				}
+			}
+
+			// If not, add it to the list of stubs
+			stub.distance = fuzzy.RankMatchNormalizedFold(domainParam, stub.name)
+			if stub.distance >= 0 {
+				stubs = append(stubs, stub)
+			}
+		}
```

Comment:
This is for ranking only; e.g. how well does the domain match the website or endpoint.